### ISSUE TITLE
debops.tinc: Improve tinc-up/tinc-down

### ISSUE THREAD

**Issue by evilham** [debops/debops#540]
Hello,

I just spent some fun time debugging networking issues :-).
(Yes, it was actually fun)

After (manually) playing quite a bit with `tinc-down` and `tinc-up`, I think they need a couple improvements:

* For starters, I guess nobody is using the (undocumented) `custom_tinc_up`. I guess that, because there is no equivalent `custom_tinc_down` but instead there is again `custom_tinc_up`. This one I think should be pretty clear: changing `tinc-down.j2` and documenting both `custom_tinc_(up|down)`. What would you think?
* Another one would be that I am missing some kind of post-what-debops-helps-me-set-up hook/variable. Something like `post_tinc_up`. I know I could use `custom_tinc_up`, but that would pretty much mean copy pasting the rest of the script and just adding the one line.
* The above probably implies that some kind of `pre_tinc_down` would be nice, for symmetry's sake.

Background: I just spent quite a bit trying to get IPv6 connectivity on an IPv4-only environment. Turns out, `tap` devices appear to have `accept_ra` set to `0`, so advertisements are ignored :-D.

Basically, I'd like to be able to add a `sysctl net.ipv6.conf.${INTERFACE}.accept_ra=1` line after everything else.

(As usual, just asking for input and would gladly implement myself!)

**Comment by drybjed** [comment]
> * For starters, I guess nobody is using the (undocumented) `custom_tinc_up`. I guess that, because there is no equivalent `custom_tinc_down` but instead there is again `custom_tinc_up`. This one I think should be pretty clear: changing `tinc-down.j2` and documenting both `custom_tinc_(up|down)`. What would you think?

Whoops, no idea how that got there, probably a copying error. Of course, switching that around and documenting is a good idea.

> * Another one would be that I am missing some kind of post-what-debops-helps-me-set-up hook/variable. Something like `post_tinc_up`. I know I could use `custom_tinc_up`, but that would pretty much mean copy pasting the rest of the script and just adding the one line.
> * The above probably implies that some kind of `pre_tinc_down` would be nice, for symmetry's sake.

Agreed, this would nicely reflect the `interfaces(5)` hook system. I wonder if adding a `run-parts /etc/network/if-up.d/` etc. with relevant variables set would be useful in tinc case...

> Background: I just spent quite a bit trying to get IPv6 connectivity on an IPv4-only environment. Turns out, `tap` devices appear to have `accept_ra` set to `0`, so advertisements are ignored :-D.
> Basically, I'd like to be able to add a `sysctl net.ipv6.conf.${INTERFACE}.accept_ra=1` line after everything else.

DebOps is supposed to be IPv6 ready, therefore this could be enabled by default in the scripts, without the need to add that specifically. I assume that with IPv4-only networks that wouldn't impact anything, no idea though about a case where IPv6 is intentionally disabled. I guess, if you check for the existence of `accept_ra` file in the `/proc` directory of the relevant interface, it should be enough to check if we can toggle it.

> (As usual, just asking for input and would gladly implement myself!)

Sounds good to me. Waiting for a PR. :-)

**Comment by evilham** [comment]
> Agreed, this would nicely reflect the interfaces(5) hook system. I wonder if adding a run-parts /etc/network/if-up.d/ etc. with relevant variables set would be useful in tinc case...

I don't think dealing with `run-parts` is particularly useful in tinc's case and the proposed vars accomplish the same :-).

> DebOps is supposed to be IPv6 ready, therefore this could be enabled by default in the scripts, without the need to add that specifically. I assume that with IPv4-only networks that wouldn't impact anything, no idea though about a case where IPv6 is intentionally disabled. I guess, if you check for the existence of accept_ra file in the /proc directory of the relevant interface, it should be enough to check if we can toggle it.

Counter proposal: a documented tinc network setting `accept_ra` that:
* If missing or True (--> defaulting to 1) or 0, 1, 2: sets that up for the specific tap-interface.
* If anything else: leave however the OS sets it up. Which probably would mean a `0` unless overridden somewhere else.

This would mean less black magic while still being perfectly IPv6 ready out of the box.


I should probably add a config example showing how to take advantage of a tinc VPN to get IPv6 connectivity on the go :-D.

**Comment by drybjed** [comment]
> I don't think dealing with `run-parts` is particularly useful in tinc's case and the proposed vars accomplish the same :-).

The hooks can do some other things, for example restart services like `dnsmasq` that bind to all interfaces, update `/etc/resolv.conf`, etc. It's not just related to the interface itself.

> Counter proposal: a documented tinc network setting `accept_ra` that:
> 
> * If missing or True (--> defaulting to 1) or 0, 1, 2: sets that up for the specific tap-interface.
> * If anything else: leave however the OS sets it up. Which probably would mean a `0` unless overridden somewhere else.
> 
> This would mean less black magic while still being perfectly IPv6 ready out of the box.

Sounds good to me.

> I should probably add a config example showing how to take advantage of a tinc VPN to get IPv6 connectivity on the go :-D.

Nice idea, go for it. :)
